internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,163 @@ import (
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
 )
 
+
+
+// Runtime2 is the VM.
+// Review: the object that implements this outlives actor calls.
+type Runtime2 struct {
+	CurerntEpoch() Epoch
+	// Review: why is this taking an offset? verify size of offset
+	Randomness(epoch Epoch, offset uint64) Randomness
+	// Review: the spec still has a TODO on this
+	// Review: not sure why the original has a messagereceipt instead of an InvocOutput
```

Comment:
I don't think it makes much of a difference, but `MessageReceipt` allows actor code to check the gas used by its subcalls, which I suppose could be a useful capability in the future. (On the other hand, we should also analyze the security implications.) @zixuanzh @jbenet Thoughts on this?

Reply:
Ok, that is interesting. So the actor might want to do something about the gas used by the functions it called, yet it has not control or visibility over its own gas consumption?
(the `Runtime` proposed has no mechanism for the actor code to consume gas).

If there is no present need for it on any of the current actors, I would suggest it is left out of the spec. The cost of changing it in the future is minimal.

Reply:
Agreed -- I think it's fine to have it return an `InvocOutput` for now